STMicroelectronics/STM32L4_L4+/app/az_ulib_dm/src/_az_ulib_dm_blob.c

```diff
@@ -174,70 +174,104 @@ static az_result copy_blob_to_flash(NXD_ADDRESS* ip, CHAR* resource, CHAR* host,
-      if ((hal_status = internal_flash_erase((UCHAR*)address, package_size)) == HAL_OK)
+      // initialize get request
+      if((nx_status = nx_web_http_client_request_initialize(&http_client,
+                                NX_WEB_HTTP_METHOD_GET, /* GET, PUT, DELETE, POST, HEAD */
+                                resource, /* "resource" (usually includes auth headers)*/
+                                host, /* "host" */
+                                0, /* Used by PUT and POST */
+                                NX_FALSE, /* If true, input_size is ignored. */
+                                NX_NULL, /* "name" */
+                                NX_NULL, /* "password" */
+                                DCF_WAIT_TIME)) == NX_SUCCESS)
       {
-        if((nx_status = nx_web_http_client_request_initialize(&http_client,
-                                  NX_WEB_HTTP_METHOD_GET, /* GET, PUT, DELETE, POST, HEAD */
-                                  resource, /* "resource" (usually includes auth headers)*/
-                                  host, /* "host" */
-                                  0, /* Used by PUT and POST */
-                                  NX_FALSE, /* If true, input_size is ignored. */
-                                  NX_NULL, /* "name" */
-                                  NX_NULL, /* "password" */
+        // add user agent
+        if((nx_status = nx_web_http_client_request_header_add(&http_client, 
+                                  USER_AGENT_NAME, 
+                                  sizeof(USER_AGENT_NAME) - 1, 
+                                  USER_AGENT_VALUE, 
+                                  sizeof(USER_AGENT_VALUE) - 1, 
                                   DCF_WAIT_TIME)) == NX_SUCCESS)
         {
-          if((nx_status = nx_web_http_client_request_header_add(&http_client, 
-                                    USER_AGENT_NAME, 
-                                    sizeof(USER_AGENT_NAME) - 1, 
-                                    USER_AGENT_VALUE, 
-                                    sizeof(USER_AGENT_VALUE) - 1, 
-                                    DCF_WAIT_TIME)) == NX_SUCCESS)
-          {
-            nx_status = nx_web_http_client_request_send(&http_client, DCF_WAIT_TIME);
-            // Download destination pointer, increase after every packet is stored
-            uint32_t total_downloaded_size = 0;
-            while(nx_status == NX_SUCCESS)
+          // send request
+          if((nx_status = nx_web_http_client_request_send(&http_client, 
+                                                          DCF_WAIT_TIME)) == NX_SUCCESS)
             {
+              // grab first chunk
               nx_status = nx_web_http_client_response_body_get(&http_client, &packet_ptr, 500);
-              if((nx_status == NX_SUCCESS) || (nx_status == NX_WEB_HTTP_GET_DONE))
+              if ((nx_status == NX_SUCCESS) || (nx_status == NX_WEB_HTTP_GET_DONE))
```

Comment:
>nx_status == NX_WEB_HTTP_GET_DONE [](http://example.com/codeflow?start=48&length=33)

If nx_status is `NX_WEB_HTTP_GET_DONE` and erase of write in flash failed, the function will return the wrong error code. So, on those situations, nx_status shall be converted to `NX_SUCCESS`.